complete tests list

yurunhub opened the pull request.

Commit:
complete tests list

Commit:
test commit

lastperson commented on test/debts.js:
```diff
@@ -47,13 +47,62 @@ contract('Debts', function(accounts) {
+    return Promise.resolve()
+    .then(() => debts.borrow(value, {from: borrower}))
+    .then(() => debts.debts(borrower))
+    .then(asserts.equal(1000));
```
You already have value const, maybe use it here?

lastperson reviewed (approved):
What should happen on zero borrow or repay?

Commit:
add new contract and tests for it

Commit:
fixes about 0 borrowing/repaying in Lender contract

yurunhub commented:
>What should happen on zero borrow or repay?

Per my understanding there should not be an Event emitted, however Depts.sol contract would not pass this check so I only added the check in my Lender.sol contract and tests

lastperson commented on test/lender.js:
```diff
@@ -0,0 +1,125 @@
+    .then(asserts.equal(0));
+  });
+
+  it('should fail on overflow when requesting money', () => {
```
is there an overflow or something else?

lastperson commented on test/lender.js:
```diff
@@ -0,0 +1,125 @@
+    .then(() => lender.requestMoney(value, {from: borrower}))
+    .then(() => asserts.throws(lender.returnMoney(0, {from: borrower})));
+  });
+
```
test for borrowing 50000000 and then another 50000001.

lastperson reviewed (changes requested):
All the functions call results should indicate to caller if they went alright or failed. In your case functions with onlyOwner/onlyNotOwner silently fail in modifier, and caller cannot verify if it failed or not.

Commit:
fix issue in contract with maxLendAmount
fix issue and rename one check from overflow to maxLendAmount test

Commit:
add return bool to public functions

yurunhub committed:
XO game homework with tests

yurunhub committed:
use evm_increaseTime instead of sleep
introduce noPenalty timeout and penalty
shorten contract code
use require instead of revert
add events per move and per penalty

yurunhub committed:
empty commit for travis

yurunhub committed:
replace Penalty system with FairPlayer motivation system
